.cargo/config.toml

```
@@ -1,4 +1,4 @@
 [env]
 # tree-sitter build fails with newer version of clang unless implicit-function-declaration is ignored
-CC_wasm32_wasi = { value = ".cargo/wasi-sdk-21.0/bin/clang -Wno-error=implicit-function-declaration", relative = true }
-AR_wasm32_wasi = { value = ".cargo/wasi-sdk-21.0/bin/ar", relative = true }
+CC_wasm32_wasi = { value = ".cargo/wasi-sdk-24.0-x86_64-linux/bin/clang -Wno-error=implicit-function-declaration", relative = true }
+AR_wasm32_wasi = { value = ".cargo/wasi-sdk-24.0-x86_64-linux/bin/ar", relative = true }
```

This path can't be os/arch specific to make sure it builds on other configs, so the tar extraction needs to be changed to be generic

```suggestion
CC_wasm32_wasi = { value = ".cargo/wasi-sdk-24.0/bin/clang -Wno-error=implicit-function-declaration", relative = true }
AR_wasm32_wasi = { value = ".cargo/wasi-sdk-24.0/bin/ar", relative = true }
```

---

I tried this change, but it fails CI. I'm trying another approach to make it generic.